ladybug/datatype/base.py

```diff
@@ -298,6 +305,15 @@ def ToString(self):
         """Overwrite .NET ToString."""
         return self.__repr__()
 
+    def __hash__(self):
+        return hash(self.__key())
```

Comment:
This is going to fail for some data types since you aren't making sure that all of the elements of the `__key` are immutable. For one, the `_unit_descr` is a dictionary and you would need to sort it by keys and turn it into two tuples in order to be able to `__hash__` it. So I say that you should remove the `__hash__` method unless you are going to do it correctly.